src/main/java/seedu/moneygowhere/storage/LocalStorage.java

```diff
@@ -6,146 +6,204 @@
 import seedu.moneygowhere.data.expense.ExpenseManager;
 import seedu.moneygowhere.exceptions.LocalStorageLoadDataInputError;
 
+import static seedu.moneygowhere.common.Configurations.LOCAL_STORAGE_DIRECTORY;
+import static seedu.moneygowhere.common.Configurations.LOCAL_STORAGE_DATA_FILE_PATH;
+import static seedu.moneygowhere.storage.LocalStorageConfigurations.XML_ROOT;
+import static seedu.moneygowhere.storage.LocalStorageConfigurations.XML_EXPENSE_ELEMENT;
+import static seedu.moneygowhere.storage.LocalStorageConfigurations.XML_SORTCONFIG_ELEMENT;
+import static seedu.moneygowhere.storage.LocalStorageConfigurations.XML_SORTCONFIG_TYPE_ATTRIBUTE;
+import static seedu.moneygowhere.storage.LocalStorageConfigurations.XML_SORTCONFIG_ORDER_ATTRIBUTE;
+import static seedu.moneygowhere.storage.LocalStorageConfigurations.XML_EXPENSE_NAME_ELEMENT;
+import static seedu.moneygowhere.storage.LocalStorageConfigurations.XML_EXPENSE_DATETIME_ELEMENT;
+import static seedu.moneygowhere.storage.LocalStorageConfigurations.XML_EXPENSE_DESCRIPTION_ELEMENT;
+import static seedu.moneygowhere.storage.LocalStorageConfigurations.XML_EXPENSE_AMOUNT_ELEMENT;
+import static seedu.moneygowhere.storage.LocalStorageConfigurations.XML_EXPENSE_AMOUNT_CURRENCY_ATTRIBUTE;
+import static seedu.moneygowhere.storage.LocalStorageConfigurations.XML_EXPENSE_CATEGORY_ELEMENT;
+import static seedu.moneygowhere.storage.LocalStorageConfigurations.XML_EXPENSE_REMARKS_ELEMENT;
+import static seedu.moneygowhere.storage.LocalStorageConfigurations.XML_EXPENSE_ID_ATTRIBUTE;
+
 import java.io.File;
 import java.io.FileNotFoundException;
-import java.io.FileWriter;
 import java.io.IOException;
 import java.math.BigDecimal;
 import java.time.LocalDateTime;
+import java.time.format.DateTimeParseException;
 import java.util.ArrayList;
-import java.util.Scanner;
 
-import static seedu.moneygowhere.common.Configurations.DIRECTORY_PATH;
-import static seedu.moneygowhere.common.Configurations.FILE_PATH_EXPENSES;
+import org.w3c.dom.Document;
+import org.w3c.dom.Element;
+import org.w3c.dom.Node;
+import org.w3c.dom.NodeList;
+import org.xml.sax.SAXException;
```

Comment:
May I check is the SAX parser built into java or is it an external library?

Reply:
Yep the SAX parser is built into java
https://docs.oracle.com/javase/7/docs/api/org/w3c/dom/package-summary.html
https://docs.oracle.com/javase/7/docs/api/javax/xml/parsers/SAXParser.html